internal/pkg/manifest/svc.go

```diff
@@ -238,11 +260,59 @@ type SidecarConfig struct {
 
 // TaskConfig represents the resource boundaries and environment variables for the containers in the task.
 type TaskConfig struct {
-	CPU       *int              `yaml:"cpu"`
-	Memory    *int              `yaml:"memory"`
-	Count     *int              `yaml:"count"` // 0 is a valid value, so we want the default value to be nil.
-	Variables map[string]string `yaml:"variables"`
-	Secrets   map[string]string `yaml:"secrets"`
+	CPU       *int                   `yaml:"cpu"`
+	Memory    *int                   `yaml:"memory"`
+	Count     AutoscalingOrTaskCount `yaml:"count"`
+	Variables map[string]string      `yaml:"variables"`
+	Secrets   map[string]string      `yaml:"secrets"`
+}
+
+// AutoscalingOrTaskCount is a custom type which supports unmarshaling yaml which
+// can either be of type int or type Autoscaling.
+type AutoscalingOrTaskCount struct {
+	TaskCount   *int // 0 is a valid value, so we want the default value to be nil.
+	Autoscaling AutoscalingConfig
+}
```

Comment:
What do you think of this?
```suggestion
type count struct {
	Value   *int // 0 is a valid value, so we want the default value to be nil.
	Autoscaling Autoscaling // Mutually exclusive with Value.
}
```

Reply:
Not able to make this one private because we need this in stack pkg unit tests